src/routine.rs

```diff
@@ -242,15 +242,31 @@ where
             elapsed_time: Duration::from_millis(0),
         };
 
-        iters
-            .iter()
-            .map(|iters| {
+        let mut results = Vec::with_capacity(iters.len());
+        results.resize(iters.len(), 0.0);
+        for (i, iters) in iters.iter().enumerate() {
+            let stack_alloc = i % 4096; // default page size
+            #[cfg(any(target_family = "unix", target_family = "windows"))]
+            {
+                alloca::with_alloca(
+                    stack_alloc, /* how much bytes we want to allocate */
+                    |_memory: &mut [core::mem::MaybeUninit<u8>] /* dynamically stack allocated slice itself */| {
```

Comment:
Maybe `stack_alloc` should be inlined here, to prevent getting a warning on platforms not selected by the `cfg` attribute. Using a name such as `_shifting_stack_space` or similar instead of `_memory` would be self-documenting.